### ISSUE TITLE
Support spaces with the Team Syncer

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/matrix-appservice-slack#657]
I'm thinking either having a Slack org bridge to a space, or have Slack channels sync to an existing space.

**Comment by tobiasBora** [comment]
Do you mean that this would sync a whole Slack Workspace (including all channels (public + private), direct messages, etc…)? This would be really cool!

**Comment by dominiwe** [comment]
If that's the case, it would probably be an awesome way for an org to _completely migrate_ from Slack to Matrix.
One could add the bridge which bridges the entire slack workspace to a space in matrix. Then one could delete the slack workspace and start adding matrix users to the rooms.

Actually, i don't really see how migration is supposed to work in another way, since slack exports for most slack users (free + pro) only contain public channels.

**Comment by tobiasBora** [comment]
Well export and sync are two different features: export would load the old history, sync would forward messages from one plateform to the other.

And using bots/custom clients, shouldn't it be possible to access private slack channels?

**Comment by dominiwe** [comment]
Yeah, sure. I think i just wanted to express that if bridging like that is possible, migration would be also.

Double puppeting is already weird imo because i guess ideally users need to be manually mapped?! And on top of that matrix doesn't really have a concept of deleting users. If an id is used up and the user disables their account the id is gone (unless manually removed from the db). It's all a little weird, which probably drives users/orgs away from migrating from slack etc to matrix even though it could be very straightforward.
